src/util.rs

```diff
@@ -1,22 +1,29 @@
-use candid::Principal;
 use serde_json::Value;
 
-pub fn to_principal(principal: &str) -> Principal {
-    match Principal::from_text(principal) {
-        Ok(p) => p,
-        Err(e) => ic_cdk::trap(&format!("failed to convert Principal {principal} {e:?}",)),
+pub fn hex_to_bytes(hex: &str) -> Option<Vec<u8>> {
+    if !hex.starts_with("0x") {
+        return None;
```

Comment:
Why is this preferable to simply cutting off `0x` if it occurs at the beginning?

Reply:
I've tried both, and this approach is more consistent with the wider Ethereum ecosystem (since JSON-RPC providers generally require the `0x` prefix). Allowing hex strings without the prefix led to confusion in the `ic-eth-starter` project, so I'd rather start this way and then relax the constraints (rather than the other way around).